lib/hourglass/redmine_patches/settings_controller_patch.rb

```diff
@@ -4,24 +4,23 @@ module SettingsControllerPatch
       extend ActiveSupport::Concern
 
       included do
-        include BooleanParsing
         alias_method_chain :plugin, :hourglass
       end
 
       def plugin_with_hourglass
         return plugin_without_hourglass unless request.post? && params[:id] == Hourglass::PLUGIN_NAME.to_s
 
-        Hourglass::Settings[] = hourglass_settings_params
-        flash[:notice] = l(:notice_successful_update)
-        redirect_to plugin_settings_path Hourglass::PLUGIN_NAME
-      end
-
-      private
-      def hourglass_settings_params
-        p = params[:settings].transform_values(&:presence)
-        p = parse_boolean [:round_default, :round_sums_only, :global_tracker], p
-        p = parse_float [:round_minimum, :round_carry_over_due], p
-        parse_int [:round_limit, :report_logo_width], p
+        settings = params[:settings].transform_values(&:presence)
+        settingsValidation = Hourglass::SettingsValidation.new settings
+        settingsValidation.global_settings = true
+        if settingsValidation.valid?
+          Hourglass::Settings[] = settings
+          flash[:notice] = l(:notice_successful_update)
+          redirect_to plugin_settings_path Hourglass::PLUGIN_NAME
+        else
+          flash[:error] = settingsValidation.errors.full_messages.to_sentence
+          redirect_to plugin_settings_path Hourglass::PLUGIN_NAME
```

Comment:
This seems odd here. The values the user has entered are lost on this redirect.
I hope there is a way to render the form settings with errors - the way usual Rails forms work.
Flashing errors is not a good practice - but I don't know how Redmine does it.